In [2]:
from scipy import stats

## Урок 4. Непрерывные случайные величины. Функция распределения и плотность распределения вероятностей. Равномерное и нормальное распределение. ЦПТ.

**1. О непрерывной равномерно распределённой случайной величине `B` известно, что её дисперсия равна 0.2. Можно ли найти правую границу величины `B` и её среднее значение зная, что левая граница равна 0.5? Если да, найдите их.**

Для равномерного распределения $D(X) = \frac{(b - a)^2}{12}$. Отсюда $b = a + \sqrt{12 \cdot D} = 1.275$. 
Среднее значение $M(X) = \frac{a + b}{2} = 2.025$

**Ответ: b=2.05, M=1.275.**

**2. Коробки с шоколадом упаковываются автоматически. Их средняя масса равна 1.06 кг. Известно, что 5% коробок имеют массу, меньшую 1 кг. Найдите: а) среднее квадратическое отклонение массы коробки, б) процент коробок, имеющих массу больше 1.1 кг.**

***Подсказка. Найдите такое значение scale, для которого значение cdf(x=1, loc=1.06, scale=scale) близко к 0.05. Точности 0.0001 будет достаточно.***

In [3]:
loc = 1.06
precision = 0.00001
p = 0
scale = 0
while p < 0.05:
    scale += precision
    p = stats.norm.cdf(1, loc, scale)
print(scale, p)

0.0364800000000005 0.05001204583680406


Получили что вес коробок представляет собой нормально распределённую случайную величину с параметрами $a = 1.06, \sigma = 0.0365$. Найдём долю коробок с массой более 1.1 кг.

In [82]:
p = 1 - stats.norm.cdf(1.1, loc, scale)
print(p)

0.13656336797447777


**Ответ: а) $\sigma = 0.0365$; б) $p = 13.66\%$**.

**3. Коробка содержит 30 конфет. Известно, что масса каждой конфеты распределена равномерно в промежутке от 12 до 14 граммов. Используя центральную предельную теорему, найти вероятность, что масса всей коробки будет: а) меньше 390 граммов, б) больше 395 граммов, в) от 380 до 400 граммов. Массой самой коробки можно пренебречь.**

В каждой коробке содержится выборка из $n=30$ конфет с равномерным распределением веса $X$ с параметрами $M(X) = 13, D(X) = \frac{(14-12)^2}{12} = \frac{1}{3}$. Тогда, согласно центральной предельной теореме, вес всей коробки будет случайной величиной с нормальным распределением с параметрами
$$a = n \cdot M(X) = 30 \cdot 13 = 390$$
$$\sigma^2 = n \cdot D(X) = 30 \cdot \frac{1}{3} = 10$$

In [83]:
loc = 390
scale = 10**0.5
F = lambda x: stats.norm.cdf(x, loc, scale)

In [84]:
# а) вес меньше 390 граммов
p1 = F(390)
print(p1)

0.5


In [85]:
# б) вес больше 395 граммов
p2 = 1 - F(395)
print(p2)

0.056923149003329065


In [87]:
# в) вес от 380 до 400 граммов
p3 = F(400) - F(380)
print(p3)

0.9984345977419975


**Ответ: а) 50%; б) 56.9%; в) 99.8%.**

**4. Каким образом связаны между собой понятие квантиля и правила двух и трёх сигм? Данное задание - на подумать, оформлять его не обязательно. Этот вопрос мы обсудим в начале следующего занятия.**